be/src/vec/common/allocator.h

```diff
@@ -146,8 +146,13 @@ class Allocator {
                                                           std::to_string(old_size) + " to " +
                                                           std::to_string(new_size) + ".",
                                                   doris::TStatusCode::VEC_CANNOT_MREMAP);
-
             /// No need for zero-fill, because mmap guarantees it.
+
+            if constexpr (mmap_populate)
+                // MAP_POPULATE seems have no effect for mremap as for mmap,
+                // Zero-fill enlarged memory range explicitly to pre-fault the pages
+                if (new_size > old_size)
+                    memset(reinterpret_cast<char*>(buf) + old_size, 0, new_size - old_size);
```

Comment:
Why manual zero-fill is needed here,

CK's comment mentions that mmap is automatically zero-filled. The manual also seems to say so: https://linux.die.net/man/2/mmap

Reply:
memset explicitly here is for pre-pault the page, not for zero fill. As I tested, MAP_POPULATE seems have no effect for mremap as for mmap.

Reply:
Thanks for the advice.